src/dataset-builder/CohortEditor.ts

```diff
@@ -592,48 +601,94 @@ export const CohortEditor: React.FC<CohortEditorProps> = (props) => {
     getNextCriteriaIndex,
   } = props;
   const [cohort, setCohort] = useState<Cohort>(originalCohort);
+  const [snapshotRequestParticipantCount, setSnapshotRequestParticipantCount] =
+    useLoadedData<SnapshotBuilderCountResponse>();
+  const defaultCohortDemographicSeries = [
+    { name: 'Asian', data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] },
+    { name: 'Black', data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] },
+    { name: 'White', data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] },
+    { name: 'Native American', data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] },
+    { name: 'Pacific Islander', data: [0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0] },
+  ];
+  const defaultCohortAgeSeries = [{ data: [0, 0, 0] }];
+  const [cohortAges, setCohortAges] = useState<CohortDemographics>(generateCohortAgeData(defaultCohortAgeSeries));
+  const [cohortDemographics, setCohortDemographics] = useState<CohortDemographics>(
+    generateCohortDemographicData(defaultCohortDemographicSeries)
+  );
+  const countStatus = snapshotRequestParticipantCount.status;
 
   const updateCohort = (updateCohort: (Cohort) => Cohort) => setCohort(updateCohort);
 
-  return h(Fragment, [
-    h(CohortEditorContents, {
-      updateCohort,
-      cohort,
-      snapshotId,
-      snapshotBuilderSettings,
-      onStateChange,
-      getNextCriteriaIndex,
-    }),
-    // add div to cover page to footer
-    div(
-      {
-        style: {
-          display: 'flex',
-          backgroundColor: editorBackgroundColor,
-          alignItems: 'end',
-          flexDirection: 'row-reverse',
-          padding: wideMargin,
+  useEffect(() => {
+    setSnapshotRequestParticipantCount(
+      withErrorReporting(`Error fetching snapshot builder count for snapshot ${snapshotId}`)(async () =>
+        DataRepo()
+          .snapshot(snapshotId)
+          .getSnapshotBuilderCount(createSnapshotBuilderCountRequest([cohort]))
+      )
+    );
+  }, [snapshotId, setSnapshotRequestParticipantCount, cohort]);
+
+  useEffect(() => {
+    countStatus === 'Ready'
+      ? (setCohortAges(generateRandomCohortAgeData()), setCohortDemographics(generateRandomCohortDemographicData()))
+      : (setCohortAges(generateCohortAgeData(defaultCohortAgeSeries)),
+        setCohortDemographics(generateCohortDemographicData(defaultCohortDemographicSeries)));
+    // @ts-ignore
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [countStatus]);
```

Comment:
this disable line is needed because, it wants a dependency on the default values, which causes a constant re-render

Reply:
by moving the default values out of the component, we don't need to include a dependency on them. and can remove this line.